src/OrchardCore/OrchardCore.Scripting.JavaScript/JavaScriptEngine.cs

```diff
@@ -23,7 +22,28 @@ public JavaScriptEngine(IMemoryCache memoryCache)
 
         public IScriptingScope CreateScope(IEnumerable<GlobalMethod> methods, IServiceProvider serviceProvider, IFileProvider fileProvider, string basePath)
         {
-            var engine = new Engine();
+            var engine = new Engine(options =>
+            {
+                options.SetWrapObjectHandler(static (e, target, type) =>
+                {
+                    if (target is JsonDynamicObject dynamicObject)
+                    {
+                        return ObjectWrapper.Create(e, (JsonObject)dynamicObject, type);
+                    }
+
+                    if (target is JsonDynamicArray dynamicArray)
+                    {
+                        return ObjectWrapper.Create(e, (JsonArray)dynamicArray, type);
+                    }
+
+                    //    if (target is JsonDynamicValue dynamicValue)
+                    //    {
+                    //        return ObjectWrapper.Create(e, dynamicValue, type);
+                    //    }
+
```

Comment:
Hi @gvkries , I found during debugging that this conversion looks unnecessary, what do you think? 

![image](https://github.com/OrchardCMS/OrchardCore/assets/15613121/4efdb47c-c799-4a29-84d6-7eaad26016fc)

Reply:
Just to be sure, I decided to convert it to `JsonValue` anyway

Reply:
I've no deep knowledge of Jint, but I guess this allows to eliminate all JSON dynamic dependencies in it all together. So I think it's worth to keep it.